File: Examples/VisualBasic/AsposePdfFacades/Images/ExtractImagesPage.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.Images
    Public Class ExtractImagesPage
        Public Shared Sub Run()
            ' ExStart:ExtractImagesPage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Images()

            ' Open input PDF
            Dim pdfExtractor As New PdfExtractor()
            pdfExtractor.BindPdf(dataDir & Convert.ToString("ExtractImages-Page.pdf"))

            ' Set StartPage and EndPage properties to the page number to
            ' You want to extract images from
            pdfExtractor.StartPage = 2
            pdfExtractor.EndPage = 2

            ' Extract images
            pdfExtractor.ExtractImage()
            ' Get extracted images
            While pdfExtractor.HasNextImage()
                ' Read image into memory stream
                Dim memoryStream As New MemoryStream()
                pdfExtractor.GetNextImage(memoryStream)

                ' Write to disk, if you like, or use it otherwise.
                Dim fileStream As New FileStream((dataDir & DateTime.Now.Ticks.ToString()) + "_out_.jpg", FileMode.Create)
                memoryStream.WriteTo(fileStream)
                fileStream.Close()
            End While
            ' ExEnd:ExtractImagesPage

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePdfFacades/Images/ExtractImagesPage.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;
using System;

namespace Aspose.Pdf.Examples.CSharp.AsposePDFFacades.Images
{
    public class ExtractImagesPage
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Images();

            //open input PDF
            PdfExtractor pdfExtractor = new PdfExtractor();
            pdfExtractor.BindPdf(dataDir+ "ExtractImages-Page.pdf");

            //set StartPage and EndPage properties to the page number to
            //you want to extract images from
            pdfExtractor.StartPage = 2;
            pdfExtractor.EndPage = 2;

            //extract images
            pdfExtractor.ExtractImage();
            //get extracted images
            while (pdfExtractor.HasNextImage())
            {
                //read image into memory stream
                MemoryStream memoryStream = new MemoryStream();
                pdfExtractor.GetNextImage(memoryStream);

                //write to disk, if you like, or use it otherwise.
                FileStream fileStream = new
                FileStream(dataDir+ DateTime.Now.Ticks.ToString() + ".jpg", FileMode.Create);
                memoryStream.WriteTo(fileStream);
                fileStream.Close();
            }
            
            
        }
    }
}